common/cmake/devices.cmake

```diff
@@ -324,19 +319,22 @@ function(DEFINE_DEVICE_TYPE)
       set(TEMP_TARGET arch.${OUTPUT_NAME}.xml)
       add_custom_command(
 	OUTPUT ${TEMP_TARGET}
-	COMMAND ${PYTHON3} ${CMD_W_ARGS} ${FINAL_OUTPUT} > ${TEMP_TARGET}
+	COMMAND ${CMD_W_ARGS} < ${FINAL_FILE} > ${TEMP_TARGET}
 	DEPENDS
         ${PYTHON3} ${PYTHON3_TARGET}
-	${CMD} ${FINAL_OUTPUT}
+	${CMD} ${FINAL_TARGET}
```

Comment:
This needs to depend on both ``${FINAL_FILE}`` and ``${FINAL_TARGET}``.  I recommend using ``append_file_dependencies`` over direct inclusion to avoid forgetting one over the other (and because of include file dependencies, if any).